### PR TITLE
refactored into controller

### LINKED ISSUE #243: [QUEST] abstract out a 'controller' layer in between routes and models
Right now routes are dealing directly with models.
As a lot of redundancy in page routes and api routes increase, we need a controller layer. (For eg `/api/users/me` and `/users/me` )

What will controllers do ? 
Controllers will supply functions like the following
(written in Typescript to show return types
```ts

function getUsers(): Promise<User[]>
function getUserById(userId: number): Promise<User>
```

Which means we can have routes simply do this - 

```js
// API routes
app.get('/api/users/:id', async (req, res) => {

	const user = await getUserById(req.params.id)
	res.json(user)

})

// Page routes
app.get('/users/:id', async (req, res) => {
	const user = await getUserById(req.params.id)
	res.render('user/id', user)
})
```

We need to start abstracting the most used models into controllers first 

- [ ] users
  - [ ] demographics (including addresses)
- [ ] clients
- [ ] authtokens

### PR DESCRIPTION
**Pull request by architkshk**
fixes #243 @championswimmer

### CODE REVIEW

File: src/controllers/clients.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+function createClient(options) {
+    return models.Client.create(options)
+}
+function updateClient(options,clientId) {
+    return models.Client.update(options, {
+        where: {id: clientId}
+    })
+}
+
+function findAllClient() {
```
Comment (championswimmer) [inline src/controllers/clients.js]: findAllClient**s** 

wherever plural return type, use plural naming

**Comment by architkshk** [comment]
@championswimmer Done!! Followed proper naming convention.

File: src/routers/pages/client.js [hunk 2]
```diff
@@ -26,29 +24,27 @@ router.get('/add',
...
             if (!client) {
                 return res.send("Invalid Client Id")
             }
             if (client.userId != req.user.id) {
                 return res.send("Unauthorized user")
             }
-
             return res.render('client/id', {client: client})
-        })
+        } catch (error) {
```
Comment (championswimmer) [inline src/routers/pages/client.js]: should do something with the error

`res.status(500).json({error: error})` at least sounds reasonable

better yet, capture in Raven too

File: src/controllers/clients.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    return models.Client.update(options, {
+        where: {id: clientId}
+    })
+}
+
+function findAllClients() {
+    return models.Client.findAll({});
+}
+
+function findAllClientsByUser(userId) {
```
Comment (championswimmer) [inline src/controllers/clients.js]: `findClientsByUserId`

File: src/controllers/demographics.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    return models.Demographic.findCreateFind({
+        where: {userId: userId},
+        include: [models.Address]
+    })
+}
+
+function createAddress(options){
+    return models.Address.create(options)
+    .then((address) => {
+        resolve(address);
```
Comment (championswimmer) [inline src/controllers/demographics.js]: what is `resolve` ???

File: src/controllers/oauth.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+function findAuthToken(clientId, userId) {
+  return models.AuthToken.findOne({
+    where: {
+      clientId: clientId,
+      userId: userId
+    }
+  });
+}
+
+function findAllAuthTokens(clientId) {
```
Comment (championswimmer) [inline src/controllers/oauth.js]: `findAuthTokensByClientId`

File: src/controllers/user.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+function findUserById(userid,includes, attributes){
+    return models.User.findOne({
+        attributes: attributes,
+        where: {id: userid},
+        include: includes
+    })
+}
+
+function UpdareUser(userid, newValues){
```
Comment (championswimmer) [inline src/controllers/user.js]: `updateUser`

File: src/oauth/oauthserver.js [hunk 7]
```diff
@@ -4,155 +4,121 @@
...
 
-const models = require('../db/models').models
-    , generator = require('../utils/generator')
+const oauthController = require('../controllers/oauth')
+    , clientController = require('../controllers/clients')
     , passport = require('../passport/passporthandler')
-    , config = require('../../config')
     , debug = require('debug')('oauth:oauthserver')
 
+const {createGrantCode,createAuthToken, findGrantCode, findAuthToken, findOrCreateAuthToken} = oauthController;
```
Comment (championswimmer) [inline src/oauth/oauthserver.js]: indent like this 

```
const {
  a,
  b,
  c,
} =require('file.js`)
```

**Comment by architkshk** [comment]
@championswimmer  Done the changes.

**Comment by architkshk** [comment]
merge master branch and resolved the conflicts.

**Comment by architkshk** [comment]
@championswimmer any updates?

**Comment by architkshk** [comment]
@championswimmer Merged master

**Pull request merged**